gradle-recommended-product-dependencies/src/main/java/com/palantir/gradle/dist/ProductDependency.java

```diff
@@ -45,22 +45,36 @@
     @JsonProperty("maximum-version")
     private String maximumVersion;
 
+    @JsonProperty("optional")
+    private boolean optional = false;
+
     public ProductDependency() {}
 
     public ProductDependency(
             String productGroup,
             String productName,
             String minimumVersion,
             String maximumVersion,
-            String recommendedVersion) {
+            String recommendedVersion,
+            boolean optional) {
         this.productGroup = productGroup;
         this.productName = productName;
         this.minimumVersion = minimumVersion;
         this.maximumVersion = maximumVersion;
         this.recommendedVersion = recommendedVersion;
+        this.optional = optional;
         isValid();
     }
 
+    public ProductDependency(
```

Comment:
I added a constructor instead of editing the existing one because it flagged further API breaks, but I'd be happy to remove it.

Reply:
This is the right approach, I believe `gradle-conjure` uses the above constructor.